src/core/server/models/user.ts

```diff
@@ -403,6 +403,10 @@ export async function insertUser(
     // Evaluate the error, if it is in regards to violating the unique index,
     // then return a duplicate Story error.
     if (err instanceof MongoError && err.code === 11000) {
+      // Check if duplicate index was about the email.
+      if (err.errmsg && err.errmsg.includes("tenantID_1_email_1")) {
```

Comment:
Might need a more concrete solution for this, as the index name isn't guaranteed.